Content.Server/StationEvents/Events/GasLeak.cs

```diff
@@ -169,16 +169,20 @@ private void Spark()
             }
         }
 
-        private bool TryFindRandomTile(out Vector2i tile)
+        public static bool TryFindRandomTile(out Vector2i tile, out StationId targetStation, out IEntity? targetGrid, out EntityCoordinates targetCoords, IRobustRandom? robustRandom = null, IEntityManager? entityManager = null)
         {
             tile = default;
+            robustRandom ??= IoCManager.Resolve<IRobustRandom>();
+            entityManager ??= IoCManager.Resolve<IEntityManager>();
 
-            _targetStation = _robustRandom.Pick(_entityManager.EntityQuery<StationComponent>().ToArray()).Station;
-            var possibleTargets = _entityManager.EntityQuery<StationComponent>()
-                .Where(x => x.Station == _targetStation).ToArray();
-            _targetGrid = _robustRandom.Pick(possibleTargets).Owner;
+            targetCoords = EntityCoordinates.Invalid;
+            targetStation = robustRandom.Pick(entityManager.EntityQuery<StationComponent>().ToArray()).Station;
+            var t = targetStation; // thanks C#
+            var possibleTargets = entityManager.EntityQuery<StationComponent>()
+                .Where(x => x.Station == t).ToArray();
+            targetGrid = robustRandom.Pick(possibleTargets).Owner;
```

Comment:
if this is gonna be static it might as well not be in this class

Reply:
Move it where?